lib/src/util.dart

```diff
@@ -270,6 +272,36 @@ final _usesOverReactRegex = RegExp(
   multiLine: true,
 );
 
+VersionRange generateNewVersionRange(
+    VersionRange currentRange, VersionRange targetRange,
+    {bool shouldOverrideLowerBound = false}) {
+  if (shouldOverrideLowerBound) {
+    return targetRange;
+  } else {
+    String versionRange;
+
+    if (currentRange.min > targetRange.min) {
+      versionRange = '>=${currentRange.min.toString()}';
+    } else {
+      versionRange = '>=${targetRange.min.toString()}';
+    }
+
+    if (targetRange.max != null) {
+      versionRange += ' <${targetRange.max.toString()}';
+    }
```

Comment:
#nit you could use the VersionRange constructor here instead of building a string to be parsed, which may not be as robust:
```dart
return VersionRange(
  min: currentRange.min > targetRange.min ? currentRange.min : targetRange.min,
  includeMin: true,
  max: targetRange.max,
);
```